### PR TITLE
Enabled .NET 4.x for project

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Added .NET 4.x support and added project dialog box to prompt users to switch.

Removed the compiler definitions for the reflection extensions.

### CODE REVIEW

File: ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -613,7 +612,7 @@ PlayerSettings:
     WebPlayer: 0
   incrementalIl2cppBuild: {}
   additionalIl2CppArgs: 
-  scriptingRuntimeVersion: 0
+  scriptingRuntimeVersion: 1
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply ProjectSettings/ProjectSettings.asset]: Oh, I thought I had added that back in the old branch, but I completely started over and forgot to cherry pick that. Thanks for reminding me. I'll be sure to get that added back in.

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply ProjectSettings/ProjectSettings.asset]: @davidkline-ms done.

**Comment by SimonDarksideJ** [comment]
So we are defintely agreed to drop 5.x support with this pr?

**Comment by StephenHodgson** [comment]
It's for the dev branch anyway, which is going to eventually be vNext. So I don't see why it should be a problem.

**Pull request merged**